crates/egui/src/input_state.rs

```diff
@@ -54,6 +54,14 @@ pub struct InputState {
     /// Position and size of the egui area.
     pub screen_rect: Rect,
 
+    /// Viewport inner position and size, only the drowable area
+    /// unit = physical pixels
+    pub inner_rect: Rect,
+
+    /// Viewport outer position and size, drowable area + decorations
+    /// unit = physical pixels
+    pub outer_rect: Rect,
```

Comment:
I suggest we combine these two into a `ViewportInfo` or similar, e.g.:

```struct
ViewportInfo {
    pub inner_rect: Option<Rect>,
    pub outer_rect: Option<Rect>,
}
```

they should be `Option`s because we don't always know where the viewport is (e.g. on web).

This would also be the natural place to add more info later, such as wether or not the viewport is maximied, focused, etc. Basically this could replace `eframe::WindowInfo`, though we don't need to do so in this PR.

Reply:
That will be nice!

So we can move the `WindowEvent::CloseRequested` to `ViewportInfo::close_requested`